tests/infra/network.py

```diff
@@ -1204,7 +1204,9 @@ def wait_for_new_primary_in(
         while time.time() < end_time:
             try:
                 logs = []
-                new_primary, new_term = self.find_primary(nodes=nodes, log_capture=logs)
+                new_primary, new_term = self.find_primary(
+                    nodes=nodes, log_capture=logs, timeout=1
```

Comment:
This is the biggest change, it means we poll through the list of nodes, and build a more up to date view. In cases where one node consumed most of the timeout, we could have run out before the global timeout, despite being in a situation where other nodes were up to date.